lms/services/canvas.py

```diff
@@ -93,7 +93,11 @@ def find_matching_file_in_course(self, course_id, file_ids):
         Return None if no matching file is found.
         """
         for file_id in file_ids:
-            file = self._file_service.get(file_id, type_="canvas_file")
+            # Use both `get` (the most common scenario, files in th same application instance)
+            # and `get_with_guid` as a fallback for installs in the same LMS with multiple application instances.
+            file = self._file_service.get(
+                file_id, type_="canvas_file"
+            ) or self._file_service.get_with_guid(file_id, type_="canvas_file")
```

Comment:
Using only the `with_guid` version should do the right thing but being extra careful here and keeping the original version for most use cases.

Reply:
I feel we should probably do the work now to work out if there's a difference. If not now then when? Otherwise I suspect we'll be carrying this complexity for a while.

These two methods are only called once here, so the "or" here could be a private matter inside the service.